src/serviceregistry/5_0_0/sd/device-discovery/device-discovery_sd.tex

```diff
@@ -0,0 +1,353 @@
+In some use cases, the topology of the Local Cloud can be important. In these situations, an application system can use the \textit{register} operation of \textbf{device-discovery} service to register the device on which the system is running. Before that, the application system should use the \textit{lookup} operation to check whether the device is already registered or not. When a device is removed from the Local Cloud, the \textit{revoke} operation should be called.
+
+\subsection{Important Delimitations}
+\label{sec:delimitations}
+
+The requester has to identify itself to use any of the operations.
+
+\subsection{Access policy}
+\label{sec:accesspolicy}
+
+Available for anyone within the local cloud.
+
+\newpage
+
+\section{Service Operations}
+\label{sec:functions}
+
+This section describes the abstract signatures of each operations of the service. The \textbf{device-discovery} service is used to \textit{register}, \textit{lookup} and \textit{revoke} devices.
+In particular, each subsection names an operation, an input type and one or two output types (unsuccessful operations can return different structure), in that order.
+The input type is named inside parentheses, while the output type is preceded by a colon. If the operation has two output types, they are separated by a slash.
+Input and output types are only denoted when accepted or returned, respectively, by the operation in question. All abstract data types named in this section are defined in Section 3.
+
+\fsubsection{register}{DeviceRegistrationRequest}{DeviceRegistrationResponse}{ErrorResponse}
+
+The registration data must meet the following criteria:
+
+\begin{itemize}
+    \item Device names are case insensitive and have to be unique within the Local Cloud.
+    \item Device names can contain maximum 63 character of letters (english alphabet), numbers and dash (-), and has to start with a letter (also cannot ends with dash).
+    \item Device has to have at least one address. It is recommended to use MAC address and/or IP address.
+    \item Keys in the metadata structure can not contain dot (.) character.
+    \item Devices can register multiple times, but only if there are identical (same addresses, same metadata).
+\end{itemize}
+
+\fsubsection{lookup}{DeviceLookupRequest}{DeviceLookupResponse}{ErrorResponse}
+
+The lookup data must meet the following criteria:
+
+\begin{itemize}
+    \item If a filter expects a list, there is an OR relation between the elements of a filter.
+    \item There is an AND relation between different kind of filters.
+\end{itemize}
+
+\fsubsection{revoke}{DeviceRevokeRequest}{OperationStatus}{ErrorResponse}
+
+The input operation data must meet the following criteria:
+
+\begin{itemize}
+    \item Devices that have system assignment can not be revoked.
+\end{itemize}
+
+
+\clearpage
+
+\section{Information Model}
+\label{sec:model}
+
+Here, all data objects that can be part of the \textbf{device-discovery} service
```

Comment:
This sentence is hard to digest. Or at least for me.... (I know it is from the previous version, but "provides to" is not a good choice in this context)
What about this way:
"Here, all data objects that can be part of the device-discovery service are listed and must be respected by the hosting System."